### DISCUSSION TITLE
implementing new slip weakening friction law (approximating TP)

### DISCUSSION THREAD

**Discussion by AliceGabriel** [SeisSol/SeisSol#1058] [Ideas]
Hello, a reviewer is asking me and @palgunadi1993 to "_quickly_" rerun a complicated model with a specific slip-weakening friction law that is currently not implemented in SeisSol. 

The form is:
shearStress = f0 normalStress / [1+slip/Dc]^1/3,
where f0 = nominal friction value. 

Peeking into  https://github.com/SeisSol/SeisSol/blob/master/src/DynamicRupture/FrictionLaws/LinearSlipWeakening.h 
this may be possible? This discussion is to assess required effort/feasibility.. 

Tagging @Thomas-Ulrich @sebwolf-de 
Thanks,
Alice

**Comment by Thomas-Ulrich** [comment]
yes, you would simply need to copy the code for BiMaterialFault (as an example for a specialization of LSW) and implement the new strength formulation https://github.com/SeisSol/SeisSol/blob/master/src/DynamicRupture/FrictionLaws/LinearSlipWeakening.cpp#L22

**Comment by palgunadi1993** [comment]
Thanks Alice.
The new C++ version makes it harder for me to understand compared to the previous Fortran version. 
I think this can be possible to implement because this is not a non-linear process.

Cheers,
Kadek

**Comment by sebwolf-de** [comment]
Hi, I've implemented a sketch of this FL on branch `fancy-lsw`. You can use this extension with `FL=1058` in the parameters file. I haven't tested this any further than compiling it once.

**Reply by palgunadi1993** [reply]
Great, Thank you @sebwolf-de and @Thomas-Ulrich , that was fast! I am just about to ask file locations.

**Reply by AliceGabriel** [reply]
Thank you so much @sebwolf-de ! @palgunadi1993 any update on this? will you try this implementation?

**Comment by palgunadi1993** [comment]
This new implementation works with newly developed large-mesh and large-fault tagging. I tested with my 3D fracture network model on SuperMUC-NG. However, the new TP approximation has no comparison with other codes. The other code is only running in 2.5D (https://academic.oup.com/gji/advance-article/doi/10.1093/gji/ggae113/7632786?searchresult=1).

**Comment by Thomas-Ulrich** [comment]
Hi,
I was about the write a PR to merge this friction law (which Vincent would also like to try), and I realized that the SR is used in the friction law, and not the slip!

**Reply by Thomas-Ulrich** [reply]
Also:
```
      stateVariable[pointIndex] = std::min(
          std::fabs(this->accumulatedSlipMagnitude[ltsFace][pointIndex]) / dC[ltsFace][pointIndex],
          static_cast<real>(1.0));
```
would need to be specialized.

**Reply by Thomas-Ulrich** [reply]
I've fixed (I think) the friction law at:
https://github.com/SeisSol/SeisSol/pull/1096
(and started working on the GPU implementation for merging)

**Reply by palgunadi1993** [reply]
Hi Thomas, 
Thanks for the update. Yes I noticed it should be Slip and not Slip Rate. However, I forgot to mention it here.
